src/modules/launcher/Plugins/Microsoft.Plugin.Uri.UnitTests/UriHelper/ExtendedUriParserTests.cs

```diff
@@ -11,86 +11,160 @@ namespace Microsoft.Plugin.Uri.UnitTests.UriHelper
+        [DataRow("http://[::1]/test", true, "http://[::1]/test", null)]
+        [DataRow("http://[::1]:1234/test", true, "http://[::1]:1234/test", null)]
+        [DataRow("https://[::1]/test", true, "https://[::1]/test", null)]
+        [DataRow("https://[::1]:1234/test", true, "https://[::1]:1234/test", null)]
+
+        // Case where `domain:port`, as specified in issue #14260
+        // Assumption: Only domain with dot is accepted
+        [DataRow("example.com:80", true, "http://example.com/", null)]
+        [DataRow("example.com:80/test", true, "http://example.com/test", null)]
+        [DataRow("example.com:80/126", true, "http://example.com/126", null)]
+        [DataRow("example.com:443", true, "https://example.com/", null)]
+        [DataRow("example.com:443/test", true, "https://example.com/test", null)]
+        [DataRow("example.com:443/126", true, "https://example.com/126", null)]
+        [DataRow("google.com:91", true, "https://google.com:91/", null)]
+        [DataRow("google.com:91/test", true, "https://google.com:91/test", null)]
+        [DataRow("google.com:91/126", true, "https://google.com:91/126", null)]
+        [DataRow("test.test.test.test:952", true, "https://test.test.test.test:952/", null)]
+        [DataRow("test.test.test.test:952/test", true, "https://test.test.test.test:952/test", null)]
+        [DataRow("test.test.test.test:952/126", true, "https://test.test.test.test:952/126", null)]
+
+        // Following cases can be both interpreted as schema:path and domain:port
+        [DataRow("tel:411", true, "https://tel:411/", "tel:411")]
+        [DataRow("tel:70421567", true, null, "tel:70421567")]
+        [DataRow("tel:863-1234", true, null, "tel:863-1234")]
+        [DataRow("tel:tel:+1-201-555-0123", true, null, "tel:tel:+1-201-555-0123")]
+
+        // All following cases should be parsed as application URI
+        [DataRow("mailto:", true, null, "mailto:")]
+        [DataRow("mailto:/", false, null, null)]
+        [DataRow("ms-settings:", true, null, "ms-settings:")]
+        [DataRow("ms-settings:/", false, null, null)]
+        [DataRow("ms-settings://", false, null, null)]
+        [DataRow("ms-settings://privacy", true, null, "ms-settings://privacy/")]
+        [DataRow("ms-settings://privacy/", true, null, "ms-settings://privacy/")]
+        [DataRow("ms-settings:privacy", true, null, "ms-settings:privacy")]
+        [DataRow("ms-settings:powersleep", true, null, "ms-settings:powersleep")]
+        [DataRow("microsoft-edge:http://google.com", true, null, "microsoft-edge:http://google.com")]
+        [DataRow("microsoft-edge:https://google.com", true, null, "microsoft-edge:https://google.com")]
+        [DataRow("microsoft-edge:google.com", true, null, "microsoft-edge:google.com")]
+        [DataRow("microsoft-edge:google.com/", true, null, "microsoft-edge:google.com/")]
+        [DataRow("microsoft-edge:https://google.com/", true, null, "microsoft-edge:https://google.com/")]
```

Comment:
Duolicated. But we should test the three cases (without proto, http and https) with port too to make sure it works.

Reply:
The assumption here is that the UriBuilder won't handle double schema and check whether the inner schema matches the port provided.

Let me know if this is necessary, I will mark it as resolved for now.